src/pdm/models/serializers.py

```diff
@@ -3,10 +3,43 @@
 from typing import Any, cast
 
 import hishel
-import msgpack
 from hishel._serializers import Metadata
 from httpcore import Request, Response
 
+try:
+    import msgpack
+except ImportError:
+    import base64
+    import json
+
+    class Encoder(json.JSONEncoder):
+        bytes_ident = "PDM_BYTES_OBJECT"
+
+        def default(self, obj: Any) -> Any:
+            if isinstance(obj, bytes):
+                base64_string = base64.b64encode(obj).decode()
+                return {"type": self.bytes_ident, "val": base64_string}
+            return super().default(obj)
+
+        @classmethod
+        def object_hook(cls, obj: Any) -> Any:
+            if isinstance(obj, dict) and obj.get("type") == cls.bytes_ident:
+                val = obj.get("val")
+                if val is not None and isinstance(val, str):
+                    return base64.b64decode(val)
+            return obj
+
+    class msgpack:  # type:ignore[no-redef]
```

Comment:
This class is clever, does this ensure both forward and backward compatibility? json loads msgpack content and msgpack loads json content